core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -528,4 +535,48 @@ private static String testGCCArgument(String argument) {
         return "";
     }
 
+    private boolean objcopyExists(Map<String, String> env) {
+        // System path
+        String systemPath = env.get(PATH);
+        if (systemPath != null) {
+            String[] pathDirs = systemPath.split(File.pathSeparator);
+            for (String pathDir : pathDirs) {
+                File dir = new File(pathDir);
+                if (dir.isDirectory()) {
+                    File file = new File(dir, "objcopy");
+                    if (file.exists()) {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        log.info("Cannot find executable (objcopy) to separate symbols from executable.");
+        return false;
+    }
+
+    private void splitDebugSymbols(Path executable) {
+        Path symbols = Paths.get(String.format("%s.debug", executable.toString()));
+        objcopy("--only-keep-debug", executable.toString(), symbols.toString());
+        objcopy("--strip-debug", executable.toString());
```

Comment:
I'd suggest to run this `objcopy("--strip-debug", executable.toString());` also when no debug symbols are being requested.  The reason for this is that OpenJDK's static libraries might contain debug symbols which in turn might end up in the native image binary and you could get rid of them in the produced native image by stripping them via `objcopy --strip-debug`.